instrumentation/spring/spring-web/spring-web-3.1/library/src/main/java/io/opentelemetry/instrumentation/spring/web/v3_1/HeaderUtil.java

```diff
@@ -6,26 +6,40 @@
 package io.opentelemetry.instrumentation.spring.web.v3_1;
 
 import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
 
 import java.lang.invoke.MethodHandle;
 import java.lang.invoke.MethodHandles;
 import java.lang.invoke.MethodType;
 import java.util.List;
-import java.util.Map;
+import java.util.function.Supplier;
 import javax.annotation.Nullable;
 import org.springframework.http.HttpHeaders;
 
 class HeaderUtil {
-  @Nullable private static final MethodHandle GET_HEADERS;
+  private static final MethodHandle GET_HEADERS;
 
   static {
-    // since spring web 7.0
-    GET_HEADERS = findGetHeadersMethod(MethodType.methodType(List.class, String.class, List.class));
+    GET_HEADERS =
+        nonNullHandle(
+            findGetHeadersMethod(
+                MethodType.methodType(List.class, String.class)), // since spring web 7.0
+            () ->
+                findGetHeadersMethod(
+                    MethodType.methodType(List.class, Object.class))); // before spring web 7.0
   }
 
+  private static MethodHandle nonNullHandle(
+      @Nullable MethodHandle first, Supplier<? extends MethodHandle> supplier) {
+    return (first != null)
+        ? first
+        : requireNonNull(supplier.get(), "Could not find suitable get method on HttpHeaders");
```

Comment:
You have to be really careful with throwing exceptions like this. For library instrumentations it isn't too bad, but in javaagent code you shouldn't throw exceptions like this (keep in mind that javaagent instrumentations sometimes use library instrumentations). In javaagent instrumentations we try not to break applications. If there are changes in the instrumented library that are not compatible with the instrumentation we disable the instrumentation instead of letting it fail. This is handled by muzzle. In javaagent instrumentations you should not fail if the methods that you need don't exist.